Infrastructure/Services/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Application.Interfaces.IRepositories;
+using Domain.Models;
+using Infrastructure.Data;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.Services.Repositories;
+
+public class GenericRepository<T> : IGenericRepository<T> where T : class, IEntity
```

Comment:
It's extra - to restrict T to class, if it's restricted to interface, it's already meaning being a class, because value type or struct can't implement an interface.

Reply:
when i remove a class i still have an error:
  GenericRepository.cs(12, 31): [CS0452] The type 'T' must be a reference type in order to use it as parameter 'TEntity' in the generic type or method 'DbSet<TEntity>'